pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -232,6 +251,17 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ct
 		return ctrl.Result{}, nil
 	}
 
+	// if this is a generic target, use a different reconciliation path
+	if isGenericTarget(externalSecret) {
+		// validate generic target configuration early
+		if err := r.validateGenericTarget(log, externalSecret); err != nil {
+			r.markAsFailed("invalid generic target", err, externalSecret, syncCallsError.With(resourceLabels), esv1.ConditionReasonSecretSyncedError)
```

Comment:
If the flag is disabled, this status is never persisted as the `defer` block for the status only starts way down below, on line 348

This means resources get no Status at all.

- Also - this condition could be `ResourceSkipped` or something like that, with the appropriate error added :)

Reply:
Oh damn, you're right. But this isn't a trivial fix, because I wanted to skip a bunch of things here... but I don't want to duplicate the defer. Aaaaa :D

Reply:
Actually, if I added one in `reconcileGenericTarget` I can do this anyways in here.